src/tests/ftest/util/general_utils.py

```diff
@@ -44,6 +44,60 @@ class DaosTestError(Exception):
     """DAOS API exception class."""
 
 
+def clush(hostlist, command, env=None, verbose=True):
+    """Run a command in parallel on a list of nodes using clush.
+
+    Args:
+        hostlist (list): list of nodes on which to run the command in parallel
+        command (str): the command to run in parallel
+        env (dict, optional): dictionary of environment variables to set.
+            Defaults to None.
+        verbose (bool, optional): should the command output be displayed.
+            Defaults to True.
+
+    Returns:
+        CmdResult: a CmdResult object containing the result of the command:
+            .command        command issued
+            .exit_status    return code from the command
+            .stdout         raw stdout (bytes)
+            .stderr         raw stderr (bytes)
+            .duration       duration of the command
+            .interrupted    was the command interrupted
+            .pid            pid of the command
+
+    """
+    nodeset = NodeSet.fromlist(hostlist)
+    clush_command = "clush -w {} -B -S \"{}\"".format(nodeset, command)
+    return process.run(
+        clush_command, env=env, ignore_status=True, shell=True,
+        verbose=verbose, allow_output_check="none")
+
+
+def check_file_exists(hosts, filename):
+    """Check if a specified file exist on each specified hosts.
+
+    Args:
+        hosts (list): list of hosts
+        filename (str): file to check for the existence of on each host
+
+    Returns:
+        (bool, NodeSet): A tuple of:
+            - True if the file exists on each of the hosts; False otherwise
+            - A NodeSet of hosts on which the file does not exist
+
+    """
+    missing_file = NodeSet()
+    nodeset = NodeSet.fromlist(hosts)
+
+    task = task_self()
+    task.run("test -e {}".format(filename), nodes=nodeset)
```

Comment:
Should probably quote the argument to `test -e` here to avoid problems with spaces (and other characters) in the filename.

Reply:
Agreed, good catch.